src/lib/code-sample-editor.ts

```diff
@@ -113,38 +132,50 @@ export class CodeSampleEditor extends LitElement {
     if (changedProperties.has('project')) {
       this._findProjectAndRegister();
     }
-    if (changedProperties.has('files')) {
+    if (changedProperties.has('files') || changedProperties.has('filename')) {
       this._currentFileIndex = 0;
+      if (this.filename && this.files) {
```

Comment:
I think using `indexOf()` would be nice, I'm not sure if using `.map()` would be nice or not:

```ts
this._currentFileIndex = this.files?.map((f) => f.name).indexOf(this.filename) ?? 0;
```

Reply:
Can't do exactly that because `this.filename` could be undefined, but I did something similar.